tests/adapters/base.py

```diff
@@ -30,55 +70,167 @@ def test_initialize_model_input(self, test_adapter):
         """
         Test that initalizing model input does as intended.
         """
+        set_values = [0.9, 1.1]
+        test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).set(set_values)
+
         assert not test_adapter._initialized
-        # TODO test for missing but mandatory parameter
-        test_adapter.initialize_model_input()
+        # start and end time should be set here as they're part of setting the
+        # timeseries views of emissions/concentrations/other drivers
+        test_adapter.initialize_model_input(start_time, end_time)
         assert test_adapter._initialized
+        # TODO test for start time and end time being passed properly
+        assert test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).get() == set_values
 
-    def test_initialize_model_input_non_model_parameter(self, test_adapter):
-        tname = ("junk",)
-        test_adapter._parameters.get_writable_scalar_view(tname, ("World",), "K").set(4)
-        test_adapter.initialize_model_input()
-        # TODO test that "junk" has not been used
+        # TODO test for missing but mandatory input
+        # We could turn this into an abstract method so that implementers are forced
+        # to do an assertion like the below.
+        test_adapter._output.get_writable_timeseries_view(
+            ("Emissions", "junk"),
+            ("World",),
+            "GtC/yr",
+            [-10, 0, 10],
+            ParameterType.POINT_TIMESERIES,
+        ).set(set_values)
+        # if you try to initialise and run with a timeseries that can't be used,
+        # you get a warning
+        assert raises_warning(test_adapter.initialize_model_input(
+            start_time,
+            end_time
+        ))
+
+        test_adapter._output.get_writable_timeseries_view(
+            ("Atmospheric Concentrations", "CO2"),
+            ("World",),
+            "ppm",
+            [298, 300, 310],
+            ParameterType.POINT_TIMESERIES,
+        ).set(set_values)
+        # If you initialise with a different run mode, the emissions get reset
+        # in order to avoid confusion
+        test_adapter.initialize_model_input(
+            start_time,
+            end_time,
+            run_mode="concentration driven"
```

Comment:
Should definetely be a model-specific parameter rather than a parameter of the method (this will never be called directly by the user, anyway)

Reply:
No, `run_mode` (or something similar) needs to be one of the key OpenSCM parameters and will be frequently used by users. One of the most annoying things about comparing SCMs at the moment is that they all have different default run modes. One of the most important jobs of the adapaters is to translate the `run_mode` into the right settings for the given model so it's trivial for users to run all models in 'emissions driven' or 'concentrations driven' or concentrations-emissions driven hybrid' mode at once so a like-with-like comparison can be made.

Reply:
@swillner does the reason for having this argument here make sense now?